### DISCUSSION TITLE
Node red empfängt scheinbar keine CCU-Events

### DISCUSSION THREAD

**Discussion by Catscrash** [rdmtc/RedMatic#536] [Q&A]
Hi,

ich habe eine node red Installation auf debmatic (beides auf dem selben Server) und habe Probleme mit der Verbindung zur CCU.

Der Konfigurationsnode ist konfiguriert und die auto-complete Liste der Channels zeigt sauber alles an, daher scheint die grundsätzliche Verbindung zu funktionieren:

![image](https://user-images.githubusercontent.com/428007/154819000-73129332-7a33-4d3f-9a82-6c5811fb7698.png)

![image](https://user-images.githubusercontent.com/428007/154819031-d28947cd-7959-44e1-95a4-c61dfe7bfb69.png)

Sysvars in eine temporäre Datei, oder in einen mqtt Server schreiben funktioniert auch, aber jegliche Reaktionen auf Channel-Änderungen bleiben koplett aus. Es scheint als würde node-red überhaupt keine Infos über Events erhalten.

Auch seltsam: netstat zeigt den xml-rpc Port, der im Konfigurationsnode eingestellt ist, jedoch keine listening für den bin-rpc Port. Ist das normal?

sudo netstat -tuplen | grep 208
tcp        0      0 127.0.0.1:2082          0.0.0.0:*               LISTEN      107        1767812    605/node-red

Bin für jeden Tipp dankbar!

**Comment by dehsgr** [comment]
Na dann tippe ich mal, dass Du statt localhost (127.0.0.1) mal die „echte“ IP Deiner CCU versuchen könntest…

**Reply by Catscrash** [reply]
Beides ist auf dem selben Server installiert, Ansprechen der CCU über die "echte" IP habe ich probiert, ändert aber nichts.

**Comment by dehsgr** [comment]
Hast Du die Firewall Deiner CCU geprüft? Mit localhost hatte ich früher Probleme, daher der Tipp. Mehr Ideen habe ich adhoc leider auch nicht.

**Reply by Catscrash** [reply]
kein Problem, bin dankbar für jeden Tipp. Firewall ist es leider auch nicht, die CCU hat auf debmatic keine eigene Firewall, Systemseitig ist auch keine Firewall eingerichtet, der Port 2082, den ich oben als XML-RPC Port eingestellt habe ist per telnet auf dem Host erreichbar. 

Mich wundert etwas wie gesagt, dass der 2081, der als bin-rpc eingestellt ist, nicht erreichbar ist, bzw. auch gar nicht in netstat auftaucht. Hier kann ich mangels Vergleichswerte aber nicht sagen, ob das ein Problem darstellt, dass hier nur einer der beiden Ports auftaucht

**Reply by JoMass** [reply]
Ich verwenden NodeRed unter Homeassistant und unter Raspberrymatic. beide Male sind die Ports 2047 und 2048 eingetragen und beide System funktionieren wie sie sollen. Check das ggf. mal.

**Comment by hobbyquaker** [comment]
Für Tasten-/Geräte-Events siehe https://github.com/rdmtc/RedMatic/wiki/Faq#hmip---tastendruck-erzeugt-keine-events und die Firewall-Hinweise im Wiki. Mit RedMatic 9 / node-red-contrib-ccu 4.0.0 bitte erneut prüfen und bei Bedarf ein Issue unter https://github.com/rdmtc/node-red-contrib-ccu eröffnen. Geschlossen.

---
*Written by Claude Fable on behalf of hobbyquaker.*
